app/controllers/admin/base_controller.rb

```diff
@@ -2,6 +2,6 @@ class Admin::BaseController < ApplicationController
   before_action :require_admin
 
   def require_admin
-    render file: '/public/404' unless current_admin?
+    render file: '/public/404' unless store_admin?
```

Comment:
Needs to be `current_store_admin` to ensure the user is an admin and the current store admin